docs/README.md

```diff
@@ -0,0 +1,78 @@
+
+A complete migration may take a while (sometimes a multi-year effort) depending on the size of the application. In order to continue deploying an application to production while working on migrating, the best pattern is to follow is the [Strangler Fig pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/strangler-fig). This pattern allows for continual development on the old system with an incremental approach to moving forward. This document will describe how to apply that pattern to an ASP.NET app migrating towards ASP.NET Core.
+
+To jump into the process, please see the [Getting Started](getting_started.md) guide.
+
+## Migration Journey
+
+When starting a migration journey, the application will be targeting ASP.NET Framework and running on Windows with its supporting libraries:
+
+```mermaid
+flowchart LR;
+  external[External Traffic] --> framework[.NET Framework Application]
+  framework --- libraries[[Business logic]]
+```
+
+The first step is to introduce a new application based on ASP.NET Core that will become the entry point. Traffic will enter the core app and if the core app cannot match a route, it will proxy the request onto the .NET Framework Application via [YARP](https://microsoft.github.io/reverse-proxy/) and serve a response as the application has already been doing. Majority of code will continue to be in the .NET Framework Application, but the core app is now set up to start migrating routes to:
+
+```mermaid
+flowchart LR;
+  external[External Traffic] --> core[ASP.NET Core Application]
+  core --- libraries
+  core -- YARP proxy --> framework[ \n\n.NET Framework Application\n\n\n]
+  framework --- libraries[[Business logic]]
+```
+
+In order to start moving over business logic that relies on `HttpContext`, the libraries need to be built against `Microsoft.AspNetCore.SystemWebAdapters`. This will ensure that the libraries are using surface area that is available on both ASP.NET and ASP.NET Core:
+
+```mermaid
+flowchart LR;
+  external[External Traffic] --> core[ASP.NET Core Application]
+  core -- Adapters --- libraries
+  core -- YARP proxy --> framework[ \n\n.NET Framework Application\n\n\n]
+  framework --- libraries[[Business logic]]
+```
+
+At this point, the journey is to focus on moving routes over one at a time. This could be WebAPI or MVC controllers (or even a single method from a controller), ASPX pages, handlers, or some other implementation of a route. If the route is available in the core app, it will then be matched and served from there. Over time, the core app will start processing more of the routes served than the .NET Framework Application:
```

Comment:
I'm not sure if you want to include it here or not, but as routes are moved it will also be necessary to move services within the app that those endpoints use. For example, the first endpoint that gets moved will probably also involve moving some models, maybe data access stuff, etc.

As that code starts to move, developers will need to start thinking about whether they should (1) duplicate the code, (2) link the code in the new project, or (3) move the code to shared libraries. It might be nice to offer some guidance on those decisions in the docs.

Reply:
Good point. I'll put those points in, but I'll punt the decision matrix for later